metaseq/modules/transformer_layer.py

```diff
@@ -26,7 +26,7 @@ def _linear(x, weight, bias=None):
     return F.linear(x, weight, bias)
 
 
-def _ffn(x, fc1, activation_fn, fc2, dropout_module):
+def _ffn(x, fc1, activation_fn, activation_gate, fc2, dropout_module):
```

Comment:
Nit: can we push the `activation_gate` logic into `activation_fn` instead? As in, remove the need to add an additional parameter here, along with an additional if/else conditional TransformerDecoderLayer's init method?

Reason for this request is that `TransformerDecoderLayer`'s init can get polluted really quickly with a bunch of if/else statements changing what gets constructed. Minimizing the branching factor there would help keep that logic under control.